solr/8.5.1/cores/solrdata/data/articles/xslt/export-xml.xsl

```diff
@@ -0,0 +1,52 @@
+<?xml version='1.0' encoding='UTF-8'?>
+
+<!-- 
+  Simple transform of query results to export XML format
+ -->
+<xsl:stylesheet version='1.0'
+    xmlns:xsl='http://www.w3.org/1999/XSL/Transform'
+>
+
+	<xsl:output media-type="text/xml; charset=UTF-8" encoding="UTF-8"/> 
+  	<xsl:variable name="tag" select="response//str[@name = 'tag']/text()"/>
+
+	<xsl:template match='/'>
+		<doc-list>
+			<xsl:apply-templates select="response/result/doc"/>
+		</doc-list>
+	</xsl:template>
+  
+	<xsl:template match="doc">		
+    	<doc>
+			<xsl:apply-templates/>
+			<xsl:if test="$tag != ''">
+				<field name="tag"><xsl:value-of select="$tag"/></field>
+			</xsl:if>
+    	</doc>
+	</xsl:template>
+
+	<xsl:template match="doc/*">
+		<field name="{@name}">
+          <xsl:value-of select="."/>
+		</field>
+	</xsl:template>
+
+	<xsl:template match="doc/*[*]">
+		<xsl:apply-templates>
+			<xsl:with-param name="name"><xsl:value-of select="@name"/></xsl:with-param>
+		</xsl:apply-templates>
+	</xsl:template>
+
+	
+	<xsl:template match="doc/*/*">	
+		<xsl:param name="name"/>
+		<field name="{$name}">
+          <xsl:value-of select="."/>
+		</field>
+	</xsl:template>
+
+	<xsl:template match="doc/*[@name='score']"/>
+
+	<xsl:template match="*"/>

```

Comment:
Que eu saiba, todos os elementos vão acabar vindo aqui e não será gerado... O último template tem prioridade sobre os demais...

Reply:
@robertatakenaka vou testar esse ponto colocado.

Reply:
@robertatakenaka pelo que estudei entendi que na XSLT existe um elemento não serializável que é referenciado pelo **XPath** '/', esse é o nó ancestral de todos os outros nós no documento XML, não tem marcação: você não pode vê-lo em um documento serializado. **É o pai do elemento mais externo!** 

Acaba que ele não é um elemento do XML e portanto se entendi corretamente não é obtido ao utilizar o **XPath** '*', portanto verifique que na linha 13: https://github.com/scieloorg/search-journals/pull/529/files#diff-271eba131665efbc4da922ab15c11a69cc8d474cb8e761e0ad408d3794e0ba2dR13, estou selecionando esse "root note" e portanto é aplicado os templates para todos os elementos abaixo dele. 

Veja as referências que li para entender isso: 

https://stackoverflow.com/questions/5952723/difference-between-content-node-and-document-root/5957159#5957159

https://stackoverflow.com/questions/21618101/difference-between-xsltemplate-match-and-xsltemplate-match

Irei remover o ``<xsl:template match="*"/>`` pois ele esta realizando um casamento desnecessário. 

Para checar o que estou comentando segue a **XSLT**, **XML** e um **script em python** para teste, é necessário a instalação do **lxml**.

[transform.zip](https://github.com/scieloorg/search-journals/files/6205983/transform.zip)

Reply:
```xml
<xsl:template match="/">
    <!-- sim isso se aplica para o elemento root -->
</xsl:template>

<xsl:template match="*">
    <!-- sim isso se aplica para qq elemento (talvez inclusive para root) -->
</xsl:template>
```
A ordem dos template match no arquivo faz diferença.
O último tem mais prioridade que o primeiro.
Então, se você tiver só isso:
```xml
<xsl:template match="doc">
    <!-- sim isso se aplica para o elemento doc -->
</xsl:template>
<xsl:template match="*">
    <!-- sim isso se aplica para qq elemento, inclusive para doc, já que está por último, mas talvez isso depende do processador-->
</xsl:template>
```
```